lib/hyacinth/adapters/digital_object_search_adapter/solr/document_generator.rb

```diff
@@ -57,29 +58,73 @@ def merge_rights_fields_for(digital_object, solr_document = {})
         end
 
         def merge_rights_fields_for!(digital_object, solr_document = {})
-          # TODO: iterate over rights fields for type
-          # build keys, inspect values
-          # add flag for category content at all
-          solr_document['copyright_status_copyright_statement_ssi'] =
-            digital_object.rights.fetch('copyright_status', [])
-                          .map { |rd| rd.dig('copyright_statement', 'pref_label') }
-                          .first
-          solr_document['copyright_status_copyright_statement_ssi'] ||= Hyacinth::DigitalObject::RightsFields::UNASSIGNED_STATUS_INDEX
+          return unless digital_object.can_have_rights?
+
+          # Add flag for rights content at all
           solr_document['rights_category_present_bi'] = digital_object.rights.present?
+
+          merge_dynamic_fields(digital_object.rights, "#{digital_object.digital_object_type}_rights", solr_document)
```

Comment:
I don't think I understand what you mean. They are configurable in the same way because rights fields are also dynamic fields. We don't expose them in the dynamic fields UI, but I think at some point we should.

Reply:
right - since we don't really have a way to change what field they get indexed into based on the UI (yet), I'm wondering if we should just make it a completely separate search field.

Reply:
like 'identifiers', 'titles', 'keywords', 'rights' search types

Reply:
I don't think they need to be broken out into their own field. Is there a use case for searching across rights fields only? It seems like for search purposes rights and descriptive metadata should be treated the same. 

Some of the rights fields already have configuration specifying which search type they should be indexed into. For example, type of content is already keyword searchable. I asked Violeta a while ago and she specified a few. We could update the dynamic field definitions when needed manually.

Reply:
makes sense!